core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -784,6 +785,13 @@ public SmoothingModel readPhraseSuggestionSmoothingModel() throws IOException {
     }
 
     /**
+     * Reads a {@link ValueFormat} from the current stream.
+     */
+    public ValueFormat readValueFormat() throws IOException {
```

Comment:
I added serialization tests, but in a dedicated test in the same package as DocValeFormat so that I can assert on the value of pkg protected fields without exposing anything to the public